### ISSUE TITLE
LaTeX Support in mobile

### ISSUE THREAD

**Issue by Yoshi8765** [zulip/zulip-mobile#2660]
The Zulip app currently does not render LaTeX formatting correctly.

A simple example from a user in my realm:
* Correctly rendering visual of post in desktop app:
![image](https://user-images.githubusercontent.com/11256209/41207541-c6b40994-6ccc-11e8-8e68-2ec61e22c905.png)

* Incorrect render on Android:
![image](https://user-images.githubusercontent.com/11256209/41207532-bb44ec68-6ccc-11e8-86a6-a4a20f63bc64.png)

* source code:
````latex
\text{DRYLAB WEEK OF 6/10/18} 
```
```latex
General
```
- These updates will either come in the form of Hardware/Software/Simulations or Chromastat/Simulations or Chromastat/Game/Simulations

```latex
Chromastat
```
We will be meeting on Tuesday to talk about expectations, deadlines, and get started with work. We have gone in depth with how the device is built and programmed so far. 

```latex
Simulations
````

**Comment by timabbott** [comment]
What's the status on this?  I just got a report about it from a friend.

**Comment by gnprice** [comment]
There's some work in #2690 on the underlying infrastructure in the app to enable us to support this. It's a bit tricky, and it's been stalled for a while; I expect we'll return to it after we complete some other significant upgrades like #2788 .

**Comment by borisyankov** [comment]
> I expect we'll return to it

I think that day is today :) (for the PR review part)

**Comment by vladp** [comment]
Thx for taking up the challenge of figuring out how to do this for react native.

Nobody is able to solve this problem, within react-native (to add the capability to display latex or MathML).

I have been researching this on and off for a couple of months, and the only thing I could find is  that this need is listed in React-native feature request tracker:

https://react-native.canny.io/feature-requests/p/latex-or-mathml-rendering-in-reactnative

It seems like rendering latex/mathML from a Webview is the only viable path at the moment.

**Comment by jainkuniya** [comment]
hey @vladp!
Yup, we will be rendering latex in the webview itself. In fact our whole message list is webview. :)

**Comment by gnprice** [comment]
Bumping the priority because we just got another request for this feature -- with Tim's comment above, I guess that means at least 5 different people asking for it :-) , which is quite a lot.

**Comment by parthi2929** [comment]
After trying multiple platforms I just settled for zulip only because it supported latex.. now I discover it does not yet in mobile..  Please add this support ASAP.. any road map?

**Comment by berceanu** [comment]
I second this request.

**Comment by gnprice** [comment]
With #3636 landed, our app infrastructure is now in much better shape to be able to implement this feature! So I'm hoping we'll have this done sometime soon.

**Comment by thosgood** [comment]
I've just started using Zulip mobile and LaTeX rendering half works: the maths is correctly rendered, but then there's a unicode approximation that follows the rendered maths immediately afterwards.

**Comment by gnprice** [comment]
More voices of people eager for this feature, over on #3744:
> I'd just like to comment that this is a really compelling feature for the many research groups that grumble daily about Slack lacking math and language-specific code. "That all works in Zulip" is a much stronger pitch than "it works with some clients". Thanks, @ray-kraesig, for implementing, and thanks in advance for the review, @gnprice.

and

> We've been waiting for this for a long time. Looking forward to it!

**Comment by cortner** [comment]
I get the same problem as `thosgood`  on the iPad app. It is fine in the browser.

**Comment by gnprice** [comment]
LaTeX support in the mobile app is now released! The new version is 26.24.147 if you're on iOS, and 26.25.148 on Android. (They're the same app built from the same code, just different version numbers due to a hiccup in the build.)

@Yoshi8765 @parthi2929 @berceanu @thosgood @jedbrown @pyrocto @cortner and everyone else who's been wanting this -- please update, try it out, and don't hesitate to file new issues if you find something that doesn't look right :slightly_smiling_face:
